packages/atomic/src/components/facets/facet-common.ts

```diff
@@ -23,3 +30,68 @@ export interface FacetValueProps {
   part?: string;
   buttonRef?: (element?: HTMLButtonElement) => void;
 }
+
+function isCategoryFacetValueRequest(
+  value: AnyFacetValueRequest
+): value is CategoryFacetValueRequest {
+  return 'children' in value && Array.isArray(value.children);
+}
+
+function isFacetValueRequest(
+  value: AnyFacetValueRequest
+): value is FacetValueRequest | CategoryFacetValueRequest {
+  return (
+    'value' in value &&
+    typeof value.value === 'string' &&
+    !('children' in value)
+  );
+}
+
+function getSelectedCategoryFacetValueRequest(
+  ancestor: CategoryFacetValueRequest
+): CategoryFacetValueRequest | null {
+  if (ancestor.state === 'selected') {
```

Comment:
For all `=== 'selected'` checks, maybe `!== 'idle'` is best if we ever expect an `excluded` value in a near/distant future.

Reply:
if `excluded` gets added, then I believe we still want this to only return the selected value or nothing, not an excluded value.